### PR TITLE
AMP liveblog ads

### PR DESCRIPTION
**Pull request by nicl**
## What does this change?

* Don't show ads if we shouldn't (this must be a regression as I'm sure we had this originally)
* support ads on liveblogs

As part of this I've had to do a lot of refactoring as there were a lot of assumptions in existing ad code that don't relate to liveblogs (positioning, styling are the big ones).

**Doesn't yet style ads for liveblogs properly!**

### CODE REVIEW

File: packages/frontend/amp/components/Blocks.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                    edition={edition}
+                    section={section}
+                    contentType={contentType}
+                    switches={switches}
+                    commercialProperties={commercialProperties}
+                    isImmersive={false}
+                    shouldHideAds={true} // always no ads *within* blocks for liveblogs
+                />
+                {block.lastUpdatedDisplay && (
+                    <span>{block.lastUpdatedDisplay}</span>
```
Comment (gtrufitt) [inline packages/frontend/amp/components/Blocks.tsx]: Could use [time element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time)?

File: packages/frontend/amp/components/Blocks.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                )}
+            </div>
+        );
+    });
+
+    if (shouldHideAds) {
+        return <>liveBlogBlocks</>;
```
Comment (gtrufitt) [inline packages/frontend/amp/components/Blocks.tsx]: Not tested locally, but is this returning JSX anyway, so might not need the warpper?

```suggestion
        return liveBlogBlocks;
```

File: packages/frontend/amp/components/WithAds.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                    {item}
+                    {ad}
+                </>
+            );
+        }
+
+        return item;
+    });
+
+    return <>{withAds}</>;
```
Comment (gtrufitt) [inline packages/frontend/amp/components/WithAds.tsx]: Again, maybe withAds is returning a jsx element already so don't need extra wrappers?

```suggestion
    return withAds;
```

File: packages/frontend/amp/lib/find-adslots.ts [hunk 4]
```diff
@@ -139,3 +140,14 @@ export const findAdSlots = (elements: CAPIElement[]): number[] => {
     }
     return adSlots;
 };
+
+// Returns index of items to place ads *after*
+export const findBlockAdSlots = (blocks: any[]): number[] => {
+    const maxAds = 8;
+
+    return blocks
```
Comment (gtrufitt) [inline packages/frontend/amp/lib/find-adslots.ts]: This is a little bit dense to parse. Can we reduce the terseness or add comments so it's easier to comprehend?

File: packages/frontend/amp/components/Blocks.tsx (hunk 2 above)
Comment (nicl) [reply packages/frontend/amp/components/Blocks.tsx]: this is a bug in fact, which I've just fixed. really we need `<>{liveBlogBlocks}</>`

File: packages/frontend/amp/components/WithAds.tsx (hunk 3 above)
Comment (nicl) [reply packages/frontend/amp/components/WithAds.tsx]: Arrays themselves aren't valid React.SFC types it seems so I think the fragment-wrapping is required.

File: packages/frontend/amp/lib/find-adslots.ts (hunk 4 above)
Comment (nicl) [reply packages/frontend/amp/lib/find-adslots.ts]: Yeah, definitely. I've added a comment.

**Pull request merged**